test/functional/ui/statusline_spec.lua

```diff
@@ -0,0 +1,317 @@
+local helpers = require('test.functional.helpers')(after_each)
+local Screen = require('test.functional.ui.screen')
+local clear = helpers.clear
+local command = helpers.command
+local feed = helpers.feed
+local eq = helpers.eq
+local funcs = helpers.funcs
+local meths = helpers.meths
+local exec = helpers.exec
+local exec_lua = helpers.exec_lua
+local eval = helpers.eval
+
+describe('statusline clicks', function()
+  local screen
+
+  before_each(function()
+    clear()
+    screen = Screen.new(40, 8)
+    screen:attach()
+    command('set laststatus=2')
+    exec([=[
+      function! MyClickFunc(minwid, clicks, button, mods)
+        let g:testvar = printf("%d %d %s", a:minwid, a:clicks, a:button)
+      endfunction
+    ]=])
+  end)
+
+  it('works', function()
+    meths.set_option('statusline', 'Not clicky stuff %0@MyClickFunc@Clicky stuff%T')
+    meths.input_mouse('left', 'press', '', 0, 6, 17)
+    eq('0 1 l', eval("g:testvar"))
+    meths.input_mouse('right', 'press', '', 0, 6, 17)
+    eq('0 1 r', eval("g:testvar"))
+  end)
+
+  it('works for winbar', function()
+    meths.set_option('winbar', 'Not clicky stuff %0@MyClickFunc@Clicky stuff%T')
+    meths.input_mouse('left', 'press', '', 0, 0, 17)
+    eq('0 1 l', eval("g:testvar"))
+    meths.input_mouse('right', 'press', '', 0, 6, 17)
```

Comment:
Why does this trigger the click function?

Reply:
Because it's a mouse input on the clickable area?

Reply:
I'll create an issue later.

Reply:
#18994